### ISSUE TITLE
Code snippet screen reader issues

### ISSUE THREAD

**Issue by dakahn** [IBM/carbon-components-react#1953]
@dakahn 
AVT 3 issues found with macOS Mojave, Chrome and VoiceOver.
For the inline code snippet VO does not:
- announce what the button is for. VO should announce that it is a "copy" button.
- announce that the text is a code snippet.
- announce "copied" after the button is pressed the first time. (note: copied is announced after the button is pressed twice).

For the single and multi-line code snippet VO:
- Incorrectly announces that the code snippet is an editable text field.
- Does not announce "copied" after the button is pressed the first time. (note: copied is announced after the button is pressed twice).

_Originally posted by @snidersd in https://github.com/IBM/carbon-components/issues/1531#issuecomment-458255961_

**Comment by dakahn** [comment]
Open PR in vanilla repo [here](https://github.com/IBM/carbon-components/pull/1969)

**Comment by dakahn** [comment]
New testing -- new issues. Reopening @sniders

**Comment by dakahn** [comment]
Addressed inline issues with #2165

Multi-line code snippet issues forthcoming yall 🤠

**Comment by dakahn** [comment]
not stale ⚡️
